From 4eb72315cf532515bb4976f08f04f36ede5f8ef2 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Thu, 11 May 2023 17:11:56 +0100
Subject: [PATCH] Revert "drm_probe_helper: Cancel previous job before starting
 new one"

This reverts commit f2fd85791301b6c123cee37fae21153cb5ffef34.
---
 drivers/gpu/drm/drm_probe_helper.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/drm_probe_helper.c b/drivers/gpu/drm/drm_probe_helper.c
index e248fb99e87a..69b0b2b9cc1c 100644
--- a/drivers/gpu/drm/drm_probe_helper.c
+++ b/drivers/gpu/drm/drm_probe_helper.c
@@ -556,11 +556,9 @@ int drm_helper_probe_single_connector_modes(struct drm_connector *connector,
 		 * disable itself again.
 		 */
 		dev->mode_config.delayed_event = true;
-		if (dev->mode_config.poll_enabled) {
-			cancel_delayed_work_sync(&dev->mode_config.output_poll_work);
+		if (dev->mode_config.poll_enabled)
 			schedule_delayed_work(&dev->mode_config.output_poll_work,
 					      0);
-		}
 	}
 
 	/* Re-enable polling in case the global poll config changed. */
-- 
2.35.5

